### PR TITLE
Add documentation on new template functions for devices

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
In the linked PR we add some new template functions that allow templates access to the DeviceRegistry. While implementing this, I found the `device_entities` function/filter that was also undocumented so I added some notes on that as well.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/53131
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/configuration/templating.markdown [hunk 1]
```diff
@@ -207,6 +207,26 @@ The same thing can also be expressed as a filter:
...
+- `device_entities(device_id)` returns a list of entities that are associated with a given device ID. Can also be used as a filter.
+- `device_attr(device_id, attr_name)` returns the value of `attr_name` for the given device ID in the device registry. Not supported in [limited templates](#limited-templates).
+- `is_device_attr(device_id, attr_name, attr_value)` returns whether the value of `attr_name` for the given device ID matches `attr_value`. Not supported in [limited templates](#limited-templates).
+
+#### Devices examples
+
+{% raw %}
+
+```text
+{{ state_attr('deadbeefdeadbeefdeadbeefdeadbeef', 'manufacturer') }}  # Sony
```
Comment (frenck) [inline source/_docs/configuration/templating.markdown]: state_attr? 😉

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_docs/configuration/templating.markdown (hunk 1 above)
Comment (raman325) [reply source/_docs/configuration/templating.markdown]: doh

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Review by frenck — APPROVED** [review]
Thanks, @raman325 👍

**Pull request merged**